### PR TITLE
Adding new version of flutter_lint 2.0.1 and adding needRepaint on [AdaptiveHeightSliverPersistentHeader] widget.

### PR DESCRIPTION
**Pull request by luanbatistadev**
was added some new rules to analysis_options.yaml to dart 2.19 or higher
was added too a new parameter to AdaptiveHeightSliverPersistentHeader widget, [needRepaint] because in some case is necessary repaint widget inside it. 
thanks for your package, it help me so much. 
att, Luan Batista - Mobile Software Engineer at Evolve - Cimed Remédios S.A. - Brazil.

### CODE REVIEW

**Comment by camillobucciarelli** [comment]
Thank you for your contribution.

if you clean comments in analysis option I merge it

**Comment by luanbatistadev** [comment]
okok, i will clean it. tnx

File: lib/size_reporting_widget.dart [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-
   const SizeReportingWidget({
-    Key? key,
+    super.key,
     required this.child,
     required this.onSizeChange,
-  }) : super(key: key);
+  });
+  final Widget child;
+  final ValueChanged<Size> onSizeChange;
```
Comment (camillobucciarelli) [inline lib/size_reporting_widget.dart:10]: I prefer to maintain the order as previously, change lint if necessary

File: pubspec.yaml [hunk 2]
```diff
@@ -11,14 +11,14 @@ platforms:
   windows:
 
 environment:
-  sdk: ">=2.17.0 <3.0.0"
-  flutter: ">=3.0.0"
+  sdk: ">=2.19.1 <3.0.0"
+  flutter: ">=3.10.1"
```
Comment (camillobucciarelli) [inline pubspec.yaml:15]: Why you would like to force to flutter 3.10.1? I prefer to support the last 3 versions, you can bump to 3.3.0 at least

File: lib/overflow_widget.dart [hunk 3]
```diff
@@ -3,16 +3,15 @@ import 'package:flutter/widgets.dart';
...
-    Key? key,
+    super.key,
     required this.onSizeChange,
     required this.child,
     required this.alignment,
-  }) : super(key: key);
+  });
+  final ValueChanged<Size> onSizeChange;
+  final Widget child;
+  final Alignment alignment;
```
Comment (camillobucciarelli) [inline lib/overflow_widget.dart:14]: I prefer to maintain the order as previously, change lint if necessary

File: lib/adaptive_height_sliver_persistent_header.dart [hunk 4]
```diff
@@ -3,18 +3,19 @@ import 'package:flutter/widgets.dart';
...
     this.floating = false,
+    this.needRepaint = false,
     this.initialHeight = .0,
-  }) : super(key: key);
+  });
+  final Widget child;
+  final bool pinned;
+  final bool floating;
+  final double initialHeight;
+  final bool needRepaint;
```
Comment (camillobucciarelli) [inline lib/adaptive_height_sliver_persistent_header.dart:18]: I prefer to maintain the order as previously, change lint if necessary

File: lib/adaptive_height_delegate.dart [hunk 5]
```diff
@@ -3,15 +3,16 @@ import 'package:flutter/cupertino.dart';
...
   AdaptiveHeightDelegate({
     required this.height,
     required this.child,
+    this.needRepaint = false,
     this.onSizeChange,
   });
+  final ValueChanged<double>? onSizeChange;
+  final Widget child;
+  final bool needRepaint;
+  double height;
```
Comment (camillobucciarelli) [inline lib/adaptive_height_delegate.dart:15]: I prefer to maintain the order as previously, change lint if necessary

File: example/pubspec.yaml [hunk 6]
```diff
@@ -1,23 +1,23 @@
...
-  cupertino_icons: ^1.0.4
   persistent_header_adaptive:
     path: ../
 
 dev_dependencies:
+  flutter_lints: ^2.0.1
   flutter_test:
     sdk: flutter
-  flutter_lints: ^1.0.4
+
```
Comment (camillobucciarelli) [inline example/pubspec.yaml:20]: I prefer to maintain the order as previously, change lint if necessary

File: pubspec.yaml [hunk 7]
```diff
@@ -11,14 +11,14 @@ platforms:
...
 
 dependencies:
   flutter:
     sdk: flutter
 
 dev_dependencies:
+  flutter_lints: ^2.0.1
   flutter_test:
     sdk: flutter
```
Comment (camillobucciarelli) [inline pubspec.yaml:24]: I prefer to maintain the order as previously, change lint if necessary

**Review by camillobucciarelli — CHANGES_REQUESTED** [review]

**Pull request merged**